### PR TITLE
tidb: update docs for new transaction features in 5.0

### PR DESCRIPTION
**Pull request by Joyinqin**
### What is changed, added or deleted? (Required)
Synchronize docs with latest changes in transaction features like async commit and 1PC.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/4779
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Have version specific changes 
- [ ] Might cause conflicts

### CODE REVIEW

File: tidb-configuration-file.md [hunk 1]
```diff
@@ -438,6 +438,51 @@ The Plan Cache configuration of the `PREPARE` statement.
 - The threshold of the TiKV load. If the TiKV load exceeds this threshold, more `batch` packets are collected to relieve the pressure of TiKV. It is valid only when the value of `tikv-client.max-batch-size` is greater than `0`. It is recommended not to modify this value.
 - Default value: `200`
 
+### `enable-one-pc` <!-- New in v5.0 -->
+
+- Specifies whether to use the first phase commit feature for transactions involving only one Region. Compared with the traditional two-phase commit, first phase commit can greatly reduce transaction commit latency and increase throughput.
```
Comment (sticnarf) [inline tidb-configuration-file.md:443]: It should be "one-phase commit", in contrast to the traditional "two-phase commit".

Comment (TomShawn) [inline tidb-configuration-file.md]:
```suggestion
- Specifies whether to use the one-phase commit feature for transactions that involve only one Region. Compared with the two-phase commit, the one-phase commit feature can greatly reduce the commit latency and increase throughput.
```

File: tidb-configuration-file.md [hunk 2]
```diff
@@ -438,6 +438,51 @@ The Plan Cache configuration of the `PREPARE` statement.
...
 - Default value: `200`
 
+### `enable-one-pc` <!-- New in v5.0 -->
+
+- Specifies whether to use the one-phase commit feature for transactions involving only one Region. Compared with the traditional two-phase commit, first phase commit can greatly reduce transaction commit latency and increase throughput.
+- Default value: `false`
+
+> **Warning:**
+>
+> This is still an experimental feature. It is **NOT** recommended that you use it in the production environment. Currently, some known issues are as follows:
```
Comment (TomShawn) [inline tidb-configuration-file.md:448]:
```suggestion
> This is still an experimental feature. It is **NOT** recommended that you use it in a production environment. Currently, the following issues are found, and be aware of them if you need to use this feature:
```

File: tidb-configuration-file.md [hunk 3]
```diff
@@ -438,6 +438,51 @@ The Plan Cache configuration of the `PREPARE` statement.
...
+### `enable-one-pc` <!-- New in v5.0 -->
+
+- Specifies whether to use the one-phase commit feature for transactions that involve only one Region. Compared with the two-phase commit, the one-phase commit feature can greatly reduce the commit latency and increase throughput.
+- Default value: `false`
+
+> **Warning:**
+>
+> This is still an experimental feature. It is **NOT** recommended that you use it in the production environment. Currently, some known issues are as follows:
+>
+> - This feature is temporarily incompatible with [TiCDC](/ticdc/ticdc-overview.md), and the feature might cause TiCDC to run abnormally.
```
Comment (TomShawn) [inline tidb-configuration-file.md:450]: I doubt `temporarily` has any useful meaning.
```suggestion
> - This feature is incompatible with [TiCDC](/ticdc/ticdc-overview.md) and might cause TiCDC to run abnormally.
```

File: tidb-configuration-file.md [hunk 4]
```diff
@@ -438,6 +438,51 @@ The Plan Cache configuration of the `PREPARE` statement.
...
+>
+> - This feature is temporarily incompatible with [Follower Read](/follower-read.md) and [TiFlash](/tiflash/tiflash-overview.md), and snapshot isolation cannot be guaranteed during the use.
```
Comment (TomShawn) [inline tidb-configuration-file.md:452]:
```suggestion
> - This feature is incompatible with [Follower Read](/follower-read.md) and [TiFlash](/tiflash/tiflash-overview.md), and snapshot isolation cannot be guaranteed.
```

File: tidb-configuration-file.md [hunk 5]
```diff
@@ -438,6 +438,51 @@ The Plan Cache configuration of the `PREPARE` statement.
...
+>
+> - External consistency cannot be guaranteed.
+>
+> - If the transaction commit is interrupted abnormally due to the shutdown of the TiDB machine while executing DDL operation, the data format might be incorrect.
```
Comment (TomShawn) [inline tidb-configuration-file.md:456]:
```suggestion
> - If the transaction commit is interrupted abnormally by the machine shutdown when the DDL operation is executed, the data format might be incorrect.
```

**Review by sticnarf — COMMENTED** [review]
LGTM

File: tidb-configuration-file.md [hunk 6]
```diff
@@ -438,6 +438,51 @@ The Plan Cache configuration of the `PREPARE` statement.
...
+
+## tikv-client.async-commit <!-- New in v5.0 -->
+
+### `enable`
+
+- Specifies whether to enable the `Async Commit` feature so that the second phase of the two-phase transaction commit performs asynchronously in the background. Enable this feature can reduce the latency of transaction commit. This feature is not compatible with [TiDB Binlog](/tidb-binlog/tidb-binlog-overview.md) and does not take effect when Binlog is enabled.
```
Comment (TomShawn) [inline tidb-configuration-file.md:462]:
```suggestion
- Enables or disables the async commit feature for the second phase of the two-phase transaction commit to perform asynchronously in the background. Enabling this feature can reduce the latency of transaction commit. This feature is not compatible with [TiDB Binlog](/tidb-binlog/tidb-binlog-overview.md) and does not take effect when Binlog is enabled.
```

File: tidb-configuration-file.md [hunk 7]
```diff
@@ -438,6 +438,51 @@ The Plan Cache configuration of the `PREPARE` statement.
...
+- Default value: `false`
+
+### `keys-limit`
+
+- Specifies the upper limit of the number of keys in an `Async Commit` transaction. The `Async Commit` feature is not suitable for transactions that are too large. Transactions that exceed this limit use the traditional two-phase commit.
```
Comment (TomShawn) [inline tidb-configuration-file.md:467]:
```suggestion
- Specifies the upper limit of the number of keys in an async commit transaction. The async commit feature is **NOT** suitable for transactions that are too large. Transactions that exceed this limit will use the two-phase commit.
```

File: tidb-configuration-file.md [hunk 8]
```diff
@@ -438,6 +438,51 @@ The Plan Cache configuration of the `PREPARE` statement.
...
+- Default value: `256`
+
+### `total-key-size-limit`
+
+- Specifies the upper limit of the total size of keys in an `Async Commit` transaction. The `Async Commit` feature is not suitable for transactions that are too long. Transactions that exceed this limit use the traditional two-phase commit.
```
Comment (TomShawn) [inline tidb-configuration-file.md:472]:
```suggestion
- Specifies the upper limit of the total size of keys in an async commit transaction. The async commit feature is **NOT** suitable for transactions in which the involved key ranges are too long. Transactions that exceed this limit will use the traditional two-phase commit.
```

File: tidb-configuration-file.md [hunk 9]
```diff
@@ -438,6 +438,51 @@ The Plan Cache configuration of the `PREPARE` statement.
...
+
+> **Warning:**
+>
+> This is still an experimental feature. It is **NOT** recommended that you use it in the production environment. Currently, some known issues are as follows:
+>
+> - This feature is temporarily incompatible with [Follower Read](/follower-read.md) and [TiFlash](/tiflash/tiflash-overview.md), and snapshot isolation cannot be guaranteed during the use.
+>
+> - External consistency cannot be guaranteed.
+>
+> - If the transaction commit is interrupted abnormally due to the shutdown of the TiDB machine while executing DDL operation, the data format might be incorrect.
```
Comment (TomShawn) [inline tidb-configuration-file.md:484]: See comments from L446 to L456.

Comment (Joyinqin) [reply tidb-configuration-file.md]: PTAL again, thx~

File: tidb-configuration-file.md [hunk 10]
```diff
@@ -438,6 +438,51 @@ The Plan Cache configuration of the `PREPARE` statement.
...
+>
+> This is still an experimental feature. It is **NOT** recommended that you use it in a production environment. Currently, the following issues are found, and be aware of them if you need to use this feature:
+>
+> - This feature is incompatible with [TiCDC](/ticdc/ticdc-overview.md) and might cause TiCDC to run abnormally.
+>
+> - This feature is incompatible with [Follower Read](/follower-read.md) and [TiFlash](/tiflash/tiflash-overview.md), and snapshot isolation cannot be guaranteed.
+>
+> - External consistency cannot be guaranteed.
+>
+> - If the transaction commit is interrupted abnormally by the machine shutdown when the DDL operation is executed, the data format might be incorrect.
```
Comment (sticnarf) [inline tidb-configuration-file.md:456]: I prefer "crash" to "shutdown".

Comment (TomShawn) [inline tidb-configuration-file.md:456]:
```suggestion
> - If the transaction commit is interrupted abnormally by the machine crash when the DDL operation is executed, the data format might be incorrect.
```

File: tidb-configuration-file.md [hunk 11]
```diff
@@ -438,6 +438,51 @@ The Plan Cache configuration of the `PREPARE` statement.
...
+
+> **Warning:**
+>
+> This is still an experimental feature. It is **NOT** recommended that you use it in a production environment. Currently, the following issues are found, and be aware of them if you need to use this feature:
+>
+> - This feature is incompatible with [Follower Read](/follower-read.md) and [TiFlash](/tiflash/tiflash-overview.md), and snapshot isolation cannot be guaranteed.
+>
+> - External consistency cannot be guaranteed.
+>
+> - If the transaction commit is interrupted abnormally by the machine shutdown when the DDL operation is executed, the data format might be incorrect.
```
Comment (TomShawn) [inline tidb-configuration-file.md:484]:
```suggestion
> - If the transaction commit is interrupted abnormally by the machine crash when the DDL operation is executed, the data format might be incorrect.
```

**Review by sticnarf — COMMENTED** [review]
LGTM

File: tidb-configuration-file.md [hunk 12]
```diff
@@ -438,6 +438,51 @@ The Plan Cache configuration of the `PREPARE` statement.
...
+- Default value: `false`
+
+### `keys-limit`
+
+- Specifies the upper limit of the number of keys in an async commit transaction. The async commit feature is **NOT** suitable for transactions that are too large. Transactions that exceed this limit will use the two-phase commit.
+- Default value: `256`
+
+### `total-key-size-limit`
+
+- Specifies the upper limit of the total size of keys in an async commit transaction. The async commit feature is **NOT** suitable for transactions in which the involved key ranges are too long. Transactions that exceed this limit will use the traditional two-phase commit.
```
Comment (TomShawn) [inline tidb-configuration-file.md:472]:
```suggestion
- Specifies the upper limit of the total size of keys in an async commit transaction. The async commit feature is **NOT** suitable for transactions in which the involved key ranges are too long. Transactions that exceed this limit will use the two-phase commit.
```

**Review by TomShawn — APPROVED** [review]
LGTM

**Pull request merged**